src/components/Form/FormRatio.react.js

```diff
@@ -0,0 +1,81 @@
+// @flow
+
+import * as React from "react";
+import cn from "classnames";
+import { Grid } from "../";
+
+type Props = {|
+  +className?: string,
+  +step?: number,
+  +min?: number,
+  +max?: number,
+  +defaultValue?: number,
+  +value?: number,
+  +onChange?: (SyntheticInputEvent<EventTarget>) => mixed,
+|};
+
+type State = {|
+  internalValue: number,
+  hasChanged: boolean,
```

Comment:
Not totally following what `hasChanged` is used for.  Can you just kind of explain the logic with `onChange` and these state variables?